lrlex/src/lib/parser.rs

```diff
@@ -13,6 +14,9 @@ lazy_static! {
         Regex::new(r"^%[sS][a-zA-Z0-9]*$").unwrap();
     static ref RE_EXCLUSIVE_START_STATE_DECLARATION: Regex =
         Regex::new(r"^%[xX][a-zA-Z0-9]*$").unwrap();
+    static ref RE_ERE_QUOTED_CHARS: Regex = Regex::new(r#"([\^|\.|\[\$|\(|\)|\*|\+|\?])"#).unwrap();
```

Comment:
Hmm, well this currently isn't quite enough to get crimson working after escaping `<`,
because it has `\{` and '\}', which aren't referenced anywhere in POSIX lex or POSIX ERE `QUOTED_CHARS`.

Simple enough to add, but I suppose I should dig through the regex crate, to figure out if there are more differences lurking about.

Reply:
I just pushed a commit to use `regex_syntax::is_meta_character`, which has some risks e.g. if `regex_syntax` decides that '<' should be a `meta_character`.  But it should be the most compatible with existing lrlex files, if we want to keep using the rust regex format over POSIX ERE.

I figured it is easy to roll back if we want to go another route, but it seems like the most maintainable/compatible option to me